[ADD] developer: add a tutorial on Owl

fdardenne opened the pull request:
This commit adds the introduction and the chapter 1 of the new
Owl/JavaScript framework tutorial.

This new tutorial allows people to discover Owl and the building blocks
of the Odoo JavaScript framework.

robodoo commented:
[Pull request status dashboard](https://mergebot.odoo.com/odoo/documentation/pull/2845).

AntoineVDV commented on content/developer/howtos/jstraining/01_owl_framework.rst:
```diff
@@ -0,0 +1,310 @@
+
+Here is an overview of what we are going to achieve in this chapter:
+
+.. image:: 01_owl_framework/overview.png
```
There is something weird going on with the images. None of them seem to be true png, but rather jpeg disguised as png :raised_eyebrow: 
```
find . -type f -name "*.png" | xargs -I{} file  -i {} | grep -v "image/png; charset=binary$"

./content/applications/productivity/iot/devices/measurement_tool/measurement_tool_03.png: image/jpeg; charset=binary
./content/developer/howtos/jstraining/05_custom_kanban_view/customerList.png: image/jpeg; charset=binary
./content/developer/howtos/jstraining/05_custom_kanban_view/customerPager.png: image/jpeg; charset=binary
./content/developer/howtos/jstraining/05_custom_kanban_view/customerSearch.png: image/jpeg; charset=binary
./content/developer/howtos/jstraining/05_custom_kanban_view/customerFilter.png: image/jpeg; charset=binary
./content/developer/howtos/jstraining/05_custom_kanban_view/customerData.png: image/jpeg; charset=binary
./content/developer/howtos/jstraining/01_owl_framework/todo.png: image/jpeg; charset=binary
./content/developer/howtos/jstraining/01_owl_framework/counter.png: image/jpeg; charset=binary
./content/developer/howtos/jstraining/01_owl_framework/card.png: image/jpeg; charset=binary
./content/developer/howtos/jstraining/01_owl_framework/overview.png: image/jpeg; charset=binary
./content/developer/howtos/jstraining/01_owl_framework/deletingTodo.png: image/jpeg; charset=binary
./content/developer/howtos/jstraining/01_owl_framework/todoList.png: image/jpeg; charset=binary
./content/developer/howtos/jstraining/01_owl_framework/createTodo.png: image/jpeg; charset=binary
./content/developer/howtos/jstraining/01_owl_framework/togglingTodo.png: image/jpeg; charset=binary
./content/developer/howtos/jstraining/02_odoo_web_framework/statistics.png: image/jpeg; charset=binary
./content/developer/howtos/jstraining/02_odoo_web_framework/newLayout.png: image/jpeg; charset=binary
./content/developer/howtos/jstraining/02_odoo_web_framework/navigationButtons.png: image/jpeg; charset=binary
./content/developer/howtos/jstraining/02_odoo_web_framework/overview_02.png: image/jpeg; charset=binary
./content/developer/howtos/jstraining/02_odoo_web_framework/pieChart.png: image/jpeg; charset=binary
./content/developer/howtos/jstraining/02_odoo_web_framework/misc.png: image/jpeg; charset=binary
```

fdardenne replied:
Screenshot for chapter1 done in the right format

AntoineVDV reviewed (changes requested):
First pass of review for the main page only.

jcs-odoo reviewed (commented):
I didn't go through the PR, but just saw that media filenames have capital letters and numbers to identify them ( e.g., `warningWidget2.png`)
I recommend following the [guidelines about media filenames](https://www.odoo.com/documentation/master/contributing/documentation/content_guidelines.html#media-files). (separate words with a hyphen) and to make sure your png files are compressed with pngquant

AntoineVDV commented:
@jcs-odoo This is still very much a WIP and will be reviewed progressively as the work on the tutorial continues ;)

fdardenne commented:
@AntoineVDV I pushed a commit where
-  I improved the introduction with a file hierarchy presentation
- I removed exercise 1.1 on the counter
- improved chapter one with a better introduction
- modified a bit the first exercise with the `Todo` component

Could you provide some feedback on it :) ?

AntoineVDV commented:
Looking good! As a next step, we could go over the training notes and select what should be kept so that exercises have a theoretical intro.

fdardenne commented:
> Looking good! As a next step, we could go over the training notes and select what should be kept so that exercises have a theoretical intro.

For the moment I introduced these notes:

Architecture->Services in chapitre 2
Network in chapitre 2
Fields in chapitre 3 
Views in chapitre 3 
Architecture in Registries dans le chapitre 3

AntoineVDV reviewed (approved):
@robodoo r+
